Warp.WebApp/Data/S3/S3FileStorage.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Amazon.Runtime.Internal;
+using Amazon.S3;
+using Amazon.S3.Model;
+using System;
+using Warp.WebApp.Models;
+
+namespace Warp.WebApp.Data.S3;
+
+public class S3FileStorage : IS3FileStorage
+{
+    public S3FileStorage(IAmazonS3Fabric amazonS3Fabric)
+    {
+        _amazonS3Fabric = amazonS3Fabric;
+    }
+
+    public async Task Save(ImageInfo imageInfo, CancellationToken cancellationToken)
+    {
+        var s3Client = _amazonS3Fabric.CreateClient();
+        using (var memoryStream = new MemoryStream(imageInfo.Content))
+        {
+            var request = new PutObjectRequest
+            {
+                BucketName = _amazonS3Fabric.GetBucketName(),
+                Key = imageInfo.Id.ToString(),
+                InputStream = memoryStream,
+                ContentType = imageInfo.ContentType
+            };
+            var result = await s3Client.PutObjectAsync(request, cancellationToken);
```

Comment:
Please, add tracing to S3 invocations. See the example at [TraceMethodMiddleware](https://github.com/punk-link/warp/pull/103/files)